components/tools/OmeroWeb/omeroweb/webclient/show.py

```diff
@@ -415,18 +416,52 @@ def initially_open_owner(self):
 def paths_to_object(conn, experimenter_id=None, project_id=None,
                     dataset_id=None, image_id=None, screen_id=None,
                     plate_id=None, acquisition_id=None, well_id=None,
-                    group_id=None):
+                    group_id=None, page_size=settings.PAGE):
+    """
+    Retrieves the parents of an object (E.g. P/D/I for image) as a list
+    of paths.
+    Lowest object in hierarchy is found by checking parameter ids in order:
+    image->dataset->project->well->acquisition->plate->screen->experimenter
+    If object has multiple paths, these can also be filtered by parent_ids.
+    E.g. paths to image_id filtered by dataset_id.
+
+    If image is in a Dataset that is paginated (imageCount > page_size)
+    then we include 'childPage', 'childCount' and 'childIndex'
+    in the dataset dict.
+
+    Note on wells:
+    Selecting a 'well' is really for selecting well_sample paths
+    if a well is specified on its own, we return all the well_sample paths
+    than match
+    """
 
-    # Set any of the parameters present and find the lowest type to find
-    # If path components are specified for incompatible paths, e.g. a dataset
-    # id and a screen id then the following priority is enforced for the
-    # object to find:
-    # image->dataset->project->well->acquisition->plate->screen->experimenter
+    qs = conn.getQueryService()
 
-    # Note on wells:
-    # Selecting a 'well' is really for selecting well_sample paths
-    # if a well is specified on its own, we return all the well_sample paths
-    # than match
+    def get_image_ids(datasetId=None, groupId=-1, ownerId=None):
```

Comment:
This should be a top level function, be documented and be integration tested individually.